src/persist-types/src/codec_impls.rs

```diff
@@ -832,7 +833,7 @@ impl ColumnRef<()> for BinaryArray {
 
 impl ColumnGet<Vec<u8>> for BinaryArray {
     fn get<'a>(&'a self, idx: usize) -> &'a [u8] {
```

Comment:
To play this back: we expect the nulls array to be present even on these non-nullable calls when they're a nullable field in some struct - but since this will only be marked null when something higher up the hierarchy is marked null, these methods should never be called with the bad indices in practice?

Reply:
> even on these non-nullable calls when they're a nullable field in some struct

Almost! It's probably easier to write out in pseudo-code:
```
column SourceData {
  column ok: Option<{  column a: String  }>,
  column err: Option<String>,
}
```
Column `'a'` of type String is non-nullable, but its parent column `'ok'` is nullable. When there is an Err we mark `'ok'` as null, and that null ends up propagating to `'a'` despite `'a'` being marked as non-nullable.

Reply:
Sorry, I meant to say "field in some nullable struct" and not "nullable field in some struct". 🤦 Pseudo-code is definitely better!